redash-searcher-sync/src/parser.rs

```diff
@@ -0,0 +1,182 @@
+//! This module is created by modifying the program at the following link: https://github.com/getredash/redash-toolbelt/blob/f6d2c40881fcacb411665c75f3afbe570533539d/redash_toolbelt/examples/find_table_names.py
+//!
+//! Copyright (c) 2020, Redash
+//! All rights reserved.
+//! About license, see: https://github.com/getredash/redash-toolbelt/blob/f6d2c40881fcacb411665c75f3afbe570533539d/LICENSE
+
+/// Format query.
+fn format_query(query: &str) -> String {
+    let stage1 = query.replace('\n', " ");
+    let re = regex::Regex::new(r"\s+").unwrap();
+    let stage2 = re.replace_all(&stage1, " ");
+    let re = regex::Regex::new(r"(\s*,\s*)").unwrap();
+    let stage3 = re.replace_all(&stage2, ",");
+
+    stage3.to_string()
+}
+
+/// Get all references from query.
+///
+/// e.g. `SELECT * FROM table1 JOIN table2 ON table1.id = table2.id`
+///     -> `["table1", "table2"]`
+pub fn parse_references_from_query(query: &str) -> Vec<String> {
```

Comment:
What happens when a query contains CTEs?
A name of a CTE should not be added to a document.
So can you add a tests for a query with CTE and update logics if the CTEs are included to the documents?

Reply:
> So can you add a tests for a query with CTE and update logics if the CTEs are included to the documents?

That's my oversight...
Thanks for your pointing, I'll fix it up.